WalletWasabi/BitcoinCore/Mempool/MempoolMirror.cs

```diff
@@ -64,55 +73,67 @@ private int AddTransactions(IEnumerable<Transaction> txs)
 		var added = 0;
 		lock (MempoolLock)
 		{
-			foreach (var tx in txs.Where(x => !Mempool.ContainsKey(x.GetHash())))
+			// Loop through the newly received transactions expect those that are already present in our mempool snapshot.
+			foreach (Transaction tx in txs.Where(x => !Mempool.ContainsKey(x.GetHash())))
```

Comment:
I think you are right. The array would be modified if Mempool would be in the linq but it is polled with ContainsKey. The exception won't come. Still, there is a difference between the two solutions because linq is lazy. Now it is checking Mempool at the beginning and set up a list. The original solution would check Mempool after every iteration. I guess both are right.